docs/design-proposals/arch/daemonset.md

```diff
@@ -0,0 +1,71 @@
+## DaemonSet in Arktos
+
+Due to the nature Arktos being a multiple-tenanted system, DaemonSet is only allowed in system tenant; only cluster admin is permitted to create/delete/update DaemonSet resources.
+
+### What is not covered yet
+At this stage of Arktos, PVC/PV for storage volume is not fully supported - out of scope of this DaemonSet design.
+
+### DaemonSet in Scale-up Arktos
+Scale-up Arktos has single master. With the admission controller blocking non-system tenant DaemonSet, Arktos is able to ensure DaemonSet proper support.
+
+### DaemonSet in Scale-out Arktos
+[Scale-out Arktos](arktos_scale_out.md) may have multiple tenant partitions and resource partitions; each of them has master components including API Server, controller manager, etc. DaemonSet resource will be managed by tenant partition, as the controllers in tenant partition have the global view of Node resources.
+
+Usually DaemonSet reuires other API resources to provision its daemon Pods with specific environmental data. For instance, configmap resoure is used to initialize as app config file, secret as app sensitive data. These supporting resources shall be at the same Tenant Partion as the DaemonSet.
+
+Different Daemonsets can reside on different Tenant Partitions. At one TP, DaemonSet should be uniquely named; however, a DaemonSet of TP could have the same name as one managed by aother TP - though this practice is not recommended.
+
+At the specific tenant partition, DaemonSet controller populates daemon Pods for the specific DaemonSet, using the global view of Nodes; its scheduler schedules each daemon Pod to the proper Node subsequently.
+
+On the node, kubelet gets notification of daemon Pod's creation, then prepares environment for the Pod, finally starts the pod in the prepared environment. The environment preparation includes fetching the specified API resources (like secret, configmap) as defined in Pod spec and provisioning in form of files/data in node's filesystem. In order to ensure API resources are fecthed from the proper tenant partition, kubelet shall maintain the mapping of daemon Pod to tenant partition origin; whenever the needed resource of the daemon Pod is fetched, kubelet shall lookup the mapping and use that of the proper tenant partition.
```

Comment:
Isn't Daemon Pod origin same as regular pod origin? Can it use existing pod origin management mechanism?

Reply:
will add a section to clarify this.